salt/kubelet/stop.sls

```diff
@@ -10,7 +10,9 @@ include:
 drain-kubelet:
   cmd.run:
     - name: |
-        kubectl --request-timeout=1m --kubeconfig={{ pillar['paths']['kubeconfig'] }} drain {{ grains['nodename'] }} --force --delete-local-data=true --ignore-daemonsets --timeout={{ pillar['kubelet']['drain-timeout'] }}s
+        # give some time to drain the node, if it times out continue to ensure the update will continue
+        # this can cause downtime of applications, so ideally an application is drainable within the drain-timeout
+        kubectl --request-timeout=1m --kubeconfig={{ pillar['paths']['kubeconfig'] }} drain {{ grains['nodename'] }} --force --delete-local-data=true --ignore-daemonsets --timeout={{ pillar['kubelet']['drain-timeout'] }}s || :
```

Comment:
I think it would be more clear if we readd the:

```
    - check_cmd:
      - /bin/true
```

that was here before.

Reply:
updated